test/test_sparse_csr.py

```diff
@@ -1307,34 +1307,62 @@ def _test_spadd_shape(nnz, shape):
 
     @dtypes(torch.float, torch.double)
     def test_mul(self, device, dtype):
+        # TODO: This whole test should be migrated to OpInfos
         def _test_spadd_shape(fn, nnz, shape):
             x = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=torch.int32)
             y = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=torch.int32)
 
-            res = fn(y, x)
+            # Forward comparison
+            res_sparse_sparse = fn(y, x)
+            res_dense_sparse = fn(y.to_dense(), x)
+            res_sparse_dense = fn(y, x.to_dense())
             expected = fn(y.to_dense(), x.to_dense()).to_sparse_csr()
-            self.assertEqual(res, expected)
+            self.assertEqual(res_sparse_sparse, expected)
+            self.assertEqual(res_dense_sparse.to_dense(), expected.to_dense())
+            self.assertEqual(res_sparse_dense.to_dense(), expected.to_dense())
```

Comment:
Why is there `to_dense` conversion?

Reply:
While the result of mul(dense, csr) is csr, the result isn't fully compressed. To achieve that we'd need to remove all zeros from the output. When converting dense to csr via sparse_mask it's not also removing all the zeros, it's just applying the indices of the given argument.

Reply:
Got it, I did not know that about `sparse_mask`. Maybe a comment could be of help.